### PR TITLE
fix: upgrade github.com/gogo/protobuf to v1.3.2 to fix CVE-2021-3121 for release-3.1

### PR DESCRIPTION
**Pull request by mainred**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
upgrade github.com/gogo/protobuf to v1.3.2 to fix CVE-2021-3121
```

### CODE REVIEW

**Comment by mainred** [comment]
@davidz627  @saad-ali could you please take a look?

File: go.mod [hunk 1]
```diff
@@ -89,3 +89,7 @@ replace k8s.io/controller-manager => k8s.io/controller-manager v0.20.0
 replace k8s.io/csi-translation-lib => k8s.io/csi-translation-lib v0.20.0
 
 replace k8s.io/mount-utils => k8s.io/mount-utils v0.20.0
+
+replace github.com/gogo/protobuf => github.com/gogo/protobuf v1.3.2
```
Comment (pohly) [inline go.mod:93]: This is not how you update dependencies. Let's also use this opportunity to update *all* dependencies to the next minor version, not just those that happen to have known issues.

Please use `go get -t -u ./... && go mod tidy && go mod vendor` and submit that.

Comment (mainred) [reply go.mod]: Thanks for pointing this out, let me re-organize other related PRs as well.
I added in the required version into the required field, then `go get -t -u ./... && go mod tidy && go mod vendor` to upgrade the mod and vendor.

File: go.mod [hunk 2]
```diff
@@ -3,39 +3,42 @@ module github.com/kubernetes-csi/external-attacher
...
 	k8s.io/csi-translation-lib v0.20.0
-	k8s.io/klog/v2 v2.4.0
+	k8s.io/klog/v2 v2.40.1
+	k8s.io/kube-openapi v0.0.0-20220124234850-424119656bbf // indirect
+	k8s.io/utils v0.0.0-20211208161948-7d6a63dca704 // indirect
+	sigs.k8s.io/structured-merge-diff/v4 v4.2.1 // indirect
+	sigs.k8s.io/yaml v1.3.0 // indirect
 )
 
+replace github.com/googleapis/gnostic v0.5.3 => github.com/googleapis/gnostic v0.5.3
```
Comment (mainred) [inline go.mod:40]: manually added this replacement to fix the namespace change from googleapis to google, https://github.com/google/gnostic/issues/262

**Comment by jsafrane** [comment]
I'd like to point out that there is no CVE in the external-attacher. CVE-2021-3121 affects only code *generated* by protobuf v1.3.1. The external-attacher does not generate such code. There may be code in `vendor/` that is vulnerable to the CVE, but then you need to update the generated code, not the generator.

In addition, external-attacher v3.1.0 just went out of [1 year of support](https://kubernetes-csi.github.io/docs/project-policies.html#support).

**Comment by mainred** [comment]
> I'd like to point out that there is no CVE in the external-attacher. CVE-2021-3121 affects only code generated by protobuf v1.3.1. The external-attacher does not generate such code. There may be code in vendor/ that is vulnerable to the CVE, but then you need to update the generated code, not the generator.

Agree, external-attacher generates no CVE, so we change only the vendored code in external-attacher, not the code outside of vendor.

>In addition, external-attacher v3.1.0 just went out of 1 year of support.

Thanks for pointing this out.

**Pull request closed without merging**